doc/design/clustered-etcd-alert-store/design.md

```diff
@@ -0,0 +1,150 @@
+# Clustered Etcd Alert Store
+
+Type: Design Document (and PR)
+
+Date: 2019-09-20
+
+Author: David C Wang <dcwangmit01@gmail.com>
+
+## Status Quo
+
+The current Alertmanager clustering implementation will allow multiple
+instances of Alertmanager to sync recently-updated notification log entries and
+alert silences via the gossip protocol.  The syncing of alert state is not yet
+implemented.
+
+Currently, Alertmanager alert state is entirely stored in process memory.  It
+is lost upon restart of Alertmanager.  This, in conjunction with the fact that
+Alertmanager alert state is not synced with clustered peers presents several
+problems:
+
+* Alert emitters must send alerts to every single Alertmanager instance in the
+  cluster.  If some Alertmanager instances receive alerts that other instances
```

Comment:
This will cause duplicate notifications, but no problems beyond that. This is considered acceptable.

Reply:
Agreed, there is nothing wrong with duplicate notifications.  The concern is that some there are corner cases where *resolved* notifications are never sent for alerts that are no longer active.

Reply:
That's also considered acceptable. There's cases where we drop firing alerts too, but at the end of the day the goal is to provide sufficiently reliable notifications to a human in a reasonable time frame when there's an issue.